src/main/java/net/sf/jabref/BibDatabaseContext.java

```diff
@@ -32,8 +32,6 @@ public BibDatabaseContext(BibDatabase database, MetaData metaData, Defaults defa
         this.defaults = Objects.requireNonNull(defaults);
         this.database = Objects.requireNonNull(database);
         this.metaData = Objects.requireNonNull(metaData);
-
-        this.setMode(getMode());
```

Comment:
Why did you delete this? What happens if you load a bib file and save it afterwards - is there a comment about the type stored in the file then still?